homeassistant/components/overkiz/cover_entities/generic_cover.py

```diff
@@ -0,0 +1,191 @@
+        if command := self.executor.select_command(*COMMANDS_CLOSE_TILT):
+            await self.executor.async_execute_command(command)
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover."""
+        if command := self.executor.select_command(*COMMANDS_STOP):
+            await self.executor.async_execute_command(command)
+
+    async def async_stop_cover_tilt(self, **kwargs: Any) -> None:
+        """Stop the cover tilt."""
+        if command := self.executor.select_command(*COMMANDS_STOP_TILT):
+            await self.executor.async_execute_command(command)
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Return if the cover is opening or not."""
+
+        if self.assumed_state:
+            return None
+
+        # Check if cover movement execution is currently running
+        if any(
+            execution.get("device_url") == self.device.device_url
+            and execution.get("command_name") in COMMANDS_OPEN + COMMANDS_OPEN_TILT
+            for execution in self.coordinator.executions.values()
+        ):
+            return True
+
+        # Check if cover is moving based on current state
+        is_moving = self.device.states.get(OverkizState.CORE_MOVING)
+        current_closure = self.device.states.get(OverkizState.CORE_CLOSURE)
+        target_closure = self.device.states.get(OverkizState.CORE_TARGET_CLOSURE)
+
+        if not is_moving or not current_closure or not target_closure:
+            return None
+
+        return cast(int, current_closure.value) > cast(int, target_closure.value)
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing or not."""
+
+        if self.assumed_state:
+            return None
+
+        # Check if cover movement execution is currently running
+        if any(
+            execution.get("device_url") == self.device.device_url
+            and execution.get("command_name") in COMMANDS_CLOSE + COMMANDS_CLOSE_TILT
+            for execution in self.coordinator.executions.values()
+        ):
+            return True
+
+        # Check if cover is moving based on current state
+        is_moving = self.device.states.get(OverkizState.CORE_MOVING)
+        current_closure = self.device.states.get(OverkizState.CORE_CLOSURE)
+        target_closure = self.device.states.get(OverkizState.CORE_TARGET_CLOSURE)
+
+        if not is_moving or not current_closure or not target_closure:
+            return None
+
+        return cast(int, current_closure.value) < cast(int, target_closure.value)
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return the device state attributes."""
+        attr = super().extra_state_attributes or {}
+
+        # Obstruction Detected attribute is used by HomeKit
+        if self.executor.has_state(OverkizState.IO_PRIORITY_LOCK_LEVEL):
+            return {**attr, **{ATTR_OBSTRUCTION_DETECTED: True}}
```

Comment:
If the state is static why do we need it?

If it's a binary state it should probably be a binary sensor instead.